eZ/Publish/Core/Repository/UserService.php

```diff
@@ -845,28 +784,30 @@ public function updateUser(APIUser $user, UserUpdateStruct $userUpdateStruct)
                 );
             }
 
-            $this->userHandler->update(
-                new SPIUser(
-                    [
-                        'id' => $loadedUser->id,
-                        'login' => $loadedUser->login,
-                        'email' => $userUpdateStruct->email ?: $loadedUser->email,
-                        'passwordHash' => $userUpdateStruct->password ?
-                            $this->createPasswordHash(
-                                $loadedUser->login,
-                                $userUpdateStruct->password,
-                                $this->settings['siteName'],
-                                $this->settings['hashType']
-                            ) :
-                            $loadedUser->passwordHash,
-                        'hashAlgorithm' => $userUpdateStruct->password ?
-                            $this->settings['hashType'] :
-                            $loadedUser->hashAlgorithm,
-                        'isEnabled' => $userUpdateStruct->enabled !== null ? $userUpdateStruct->enabled : $loadedUser->enabled,
-                        'maxLogin' => $userUpdateStruct->maxLogin !== null ? (int)$userUpdateStruct->maxLogin : $loadedUser->maxLogin,
-                    ]
-                )
-            );
+            $spiUser = new SPIUser([
+                'id' => $loadedUser->id,
+                'login' => $loadedUser->login,
+                'email' => $userUpdateStruct->email ?: $loadedUser->email,
+                'isEnabled' => $userUpdateStruct->enabled !== null ? $userUpdateStruct->enabled : $loadedUser->enabled,
+                'maxLogin' => $userUpdateStruct->maxLogin !== null ? (int)$userUpdateStruct->maxLogin : $loadedUser->maxLogin,
```

Comment:
```suggestion
                'maxLogin' => $userUpdateStruct->maxLogin ?? $loadedUser->maxLogin,
``` if we could avoid casting to `int`